aiosmtpd/handlers.py

```diff
@@ -89,15 +89,15 @@ def __init__(self, remote_hostname, remote_port):
         self._port = remote_port
 
     def process_message(self, peer, mailfrom, rcpttos, data, **kws):
-        lines = data.split('\n')
+        lines = data.splitlines(keepends=True)
         # Look for the last header
         i = 0
         for line in lines:                          # pragma: nobranch
-            if not line:
+            if line in ('\r', '\n', '\r\n'):
                 break
             i += 1
-        lines.insert(i, 'X-Peer: %s' % peer[0])
-        data = NEWLINE.join(lines)
+        lines.insert(i, 'X-Peer: %s\r\n' % peer[0])
+        data = ''.join(lines)
```

Comment:
I generally prefer to use a module constant when string joining a sequence.  How about adding the following to the top of the module and using it here, in a manner similar to `COMMASPACE` and `NEWLINE`:

```
EMPTYSTRING = ''

...

        data = EMPTYSTRING.join(lines)
```

(I know I'm weird, but I picked up this habit back when `str.join()` was first added so it's a tough habit to break.)